script/zap-submodules

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+set -e
+
+cd "${0%/*}/.."
+rm -rf vendor/grammars vendor/CodeMirror .git/modules
+{ sed -n < .git/config 's/^\[submodule *"\([^"]\{1,\}\)"\]/submodule.\1/p' \
+| xargs -n1 git config -f .git/config --remove-section
+} 2>&1 >/dev/null || :
+
+git checkout -- vendor
```

Comment:
What's this script for and any reason we couldn't use `git rm <submodule path> && git commit` or similar instead?

Reply:
It's to reset one's working tree to the same state as a freshly checked-out copy of Linguist (save for the fact there's no need to rerun `bundle install`). Basically, it ensures there are no lingering remnants of an earlier submodule install, as submodules tend to get backed up to `.git/modules` when first cloned.

This script was used to test bootstrap timings, but it may also be helpful to users whose working directory has been left in an inconsistent state after botching an install or a particularly messy rebase.